bin/pt-online-schema-change

```diff
@@ -9607,11 +9607,10 @@ sub main {
          #                   ''
          #   doesn't match '(?-xism:Failed to find a unique new table name)'
 
-         # (*) Frank:  commented them out because it caused infinite loop
-         # and the mentioned test error doesn't arise
-
          my $original_error = $EVAL_ERROR;
-         my $original_error_code = $?;
+         my $original_error_code = $? ? $? : $!;
```

Comment:
ternary operators are not recommended according to Perl Best Practices so it might be better to replace it with something like this

```
my $original_error_code;
if ($?) {
    $original_error_code = $?;
} 
else {
    $original_error_code = $!;
}
```

Reply:
`my $original_error_code;
if ( $? ) {
   $original_error_code = $?;
} 
else {
   $original_error_code = $!;
}
`